### ISSUE TITLE
Use the new string formatting method in io.ascii.write

### ISSUE THREAD

**Issue by Gabriel-p** [astropy/astropy#8020]
As far as I can see the `formats`parameter in `io.ascii.write`  [still uses](http://docs.astropy.org/en/stable/io/ascii/write.html#parameters-for-write) the old string formatting.

The new method should be used instead.

**Comment by pllim** [comment]
Hi. Can you please be more specific? Do you mean the `%12.1f` in the docstring example? http://docs.astropy.org/en/latest/io/ascii/write.html#parameters-for-write

**Comment by Gabriel-p** [comment]
Yes, that's what I mean. And I'm talking about the [new `.format()` method](https://pyformat.info/).

**Comment by taldcroft** [comment]
I have a vague recollection of a surprising result that the old style formatting was noticeably faster.  However, that is totally vague and might not be correct, but it's worth checking on the speed as well as style here.

**Comment by SG004** [comment]
I was looking at this Issue, and I am a little confused as what exactly is being requested. That said I did a little research on the types of formatting. There seems to be a few more than just the old % style and the new .format style. One of the sources I found described String Interpolation/f-strings. Would that be better instead? And what exactly is being requested?

**Comment by taldcroft** [comment]
So my vague recollection was apparently correct and the venerable `%` formatting is almost a factor of two *faster*:
```
In [15]: timeit '%12.1f' % np.pi
363 ns ± 2.22 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [16]: timeit '{:12.1f}'.format(np.pi)
637 ns ± 4.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [19]: timeit format(np.pi, '12.1f')
694 ns ± 5.09 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```
However, the actual performance when writing a table does not change *that* much between different methods, though it does get a factor of 4 slower by writing formatted values instead of unformatted (new issue coming).
```
In [21]: c = Column(np.random.uniform(size=1_000_000))

In [22]: t = Table([c, c, c])

In [23]: time t.write('junk.dat', format='ascii.fast_basic', overwrite=True)
CPU times: user 5.17 s, sys: 72 ms, total: 5.24 s
Wall time: 5.18 s

In [24]: c = Column(np.random.uniform(size=1_000_000), format='%12.8f')

In [25]: t = Table([c, c, c])

In [26]: time t.write('junk.dat', format='ascii.fast_basic', overwrite=True)
CPU times: user 21.2 s, sys: 108 ms, total: 21.3 s
Wall time: 21.3 s

In [27]: c = Column(np.random.uniform(size=1_000_000), format='12.8f')

In [28]: t = Table([c, c, c])

In [29]: time t.write('junk.dat', format='ascii.fast_basic', overwrite=True)
CPU times: user 21.8 s, sys: 67.7 ms, total: 21.9 s
Wall time: 21.8 s
```

**Comment by taldcroft** [comment]
Takeaway: feel free to change any examples or code from old style (e.g. `format='%12.1f'`) to new style (e.g. `12.1f`), which is the preferred idiom.  This uses `format(val, fmt)` to actually do the formatting.

Note that is NOT the same as `{:12.1f}`, which uses `fmt.format(val)`.

**Comment by dhomeier** [comment]
There was some recent discussion on performance implications of the different format option in context of [`logging`](https://github.com/astropy/specutils/pull/833#discussion_r659890937) which might be of interest here as well.
